test/mpi/runtests.in

```diff
@@ -877,8 +889,14 @@ sub RunMPIProgram {
             }
         }
         $rc = close ( MPIOUT );
-        my $end_time = gettimeofday();
+        my $end_time = gettimeofday();  # seconds in floating point
         $runtime = $end_time - $start_time;
+        # count # of timeout when tests are configured with sufficient timeLimit
+        if ($test_opt->{_timeout} > 60) {
+            if ($runtime - $test_opt->{_timeout} >= -10) {
```

Comment:
Just getting this straight in my head: this is considering a successful test that runs within 10 seconds of its time limit to be a timeout?

Reply:
Yes. I admit probably this is an overkill. I give it a 10 seconds buffer zone in case the clock or timeout mechanism has discrepancy with the time measured in Perl. Probably 0 instead of -10 is fine here.

Reply:
I think the later commit gives us a good enough buffer against pre-mature termination unless *most* tests are timing out. I do agree there is probably some timer discrepancy that might trip us up if we are being too precise, so overall I'm OK.